CMakeLists.txt

```diff
@@ -1,5 +1,5 @@
-cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
-  
+cmake_minimum_required(VERSION 3.11 FATAL_ERROR)
```

Comment:
I'd recommend 3.10.2 as that is what is distributed by default in Ubuntu 18.04 and is rapidly becoming a baseline standard for CI environments.